Python modules without __init__.py failed to be parsed since 2.5.0

Pylint 2.5 fails to parse modules without __init__.py while 2.4.4 works fine.
I found nothing in the changelog regarding this, so I guess it's a bug.

### Steps to reproduce
1. create a python module without  __init__.py
2. run `pylint --disable=C,R ${MODULE_NAME}`

### Current behavior
raises:
```
************* Module ${MODULE_NAME}
${MODULE_NAME}/__init__.py:1:0: F0010: error while code parsing: Unable to load file ${MODULE_NAME}/__init__.py:
[Errno 2] No such file or directory: '${MODULE_NAME}/__init__.py' (parse-error)
```

### Expected behavior
No error expected since __init__.py are optional since python 3.4+

### pylint --version output
pylint==2.5.0
python==3.6.10

--- a/ChangeLog
+++ b/ChangeLog
@@ -25,6 +25,10 @@ Release date: TBA
 
   Close #3524
 
+* Allow linting directories without `__init__.py` which was a regression in 2.5.
+
+  Close #3528
+
 What's New in Pylint 2.5.0?
 ===========================
 

--- a/pylint/utils/utils.py
+++ b/pylint/utils/utils.py
@@ -142,7 +142,7 @@ def expand_modules(files_or_modules, black_list, black_list_re):
             continue
 
         module_path = get_python_path(something)
-        additional_search_path = [module_path] + path
+        additional_search_path = [".", module_path] + path
         if os.path.exists(something):
             # this is a file or a directory
             try:

--- a/tests/test_self.py
+++ b/tests/test_self.py
@@ -763,3 +763,18 @@ class TestRunTC:
                 ],
                 code=0,
             )
+
+    def test_can_list_directories_without_dunder_init(self, tmpdir):
+        test_directory = tmpdir / "test_directory"
+        test_directory.mkdir()
+        spam_module = test_directory / "spam.py"
+        spam_module.write("'Empty'")
+
+        with tmpdir.as_cwd():
+            self._runtest(
+                [
+                    "--disable=missing-docstring, missing-final-newline",
+                    "test_directory",
+                ],
+                code=0,
+            )

